app/coffeescripts/str/TextHelper.coffee

```diff
@@ -68,7 +68,7 @@ define [
           else
             link = match
             link = "http://" + link if link[0..3] == 'www.'
-            link = encodeURI(link).replace(/'/g, '%27')
```

Comment:
Would it be more effective to always decode the URI first, then encode it again? You have to get back to a good baseline, and taking the URI back to the baseline of totally decoded, you can then encode it again, no?